etcdserver/api/v3rpc/watch.go

```diff
@@ -66,6 +70,15 @@ func (ws *watchServer) Watch(stream pb.Watch_WatchServer) error {
 	return sws.recvLoop()
 }
 
+func (sws *serverWatchStream) newResponseHeader() *pb.ResponseHeader {
+	return &pb.ResponseHeader{
+		ClusterId: uint64(sws.server.ClusterID()),
+		MemberId:  uint64(sws.server.ID()),
+		Revision:  sws.watchStream.Revision(),
```

Comment:
We want to attach the revision when this response is generated rather than when we return this event to the user.

Say we generate a response inside storage pkg at revision 1000. When you get revision here, the revision might increase to 1002.